### DISCUSSION TITLE
globally change font

### DISCUSSION THREAD

**Discussion by sancelot** [primefaces/community#3242] [PrimeReact] [answered]
Hi,

Is there a way to globally change the font of primereact components ?

I tried, but primereact components does not follow my globally font applied 

Thanks ^^

**Comment by PH-Codestone** [comment]
this sounds like a stylesheet thing. My code has 
**import 'primereact/resources/themes/blah blah blah.css'**
way up in app.js, so it's enclosing all the components below.

**Reply by melloware** [reply]
@sancelot if you bring up the PR showcase options and click on Scale you can see it dynamically changes the font size of the whole app.

![image](https://github.com/user-attachments/assets/659c948d-d88d-4537-842e-887b2cb6c9ff)


And its documented here: https://primereact.org/theming/#scale

**Comment by sancelot** [comment]
I assume you can change scaling .... but that s not the question I asked.

I want to change the font type globally used .

in css style , if I use similar implementation as above , all the pure HTML components I implement are using the global css font, 
but any primereact component does not use it.

  
/* global font css */
@font-face {
  font-family: "Mulish";
  font-style: normal;
  font-weight: 200 1000;
  font-display: block;
  src: local("Mulish"), url("./assets/badab.ttf") format("truetype");
}
html,
body,
#root {
  --main-blue: #014f7e;
  --main-grey: #ecf0f1;
  font-family: "Mulish", sans-serif;
}

**Comment by melloware** [comment]
is it being overridden by the theme somehow?  Change the `$fontFamily` variable in your theme?

**Comment by sancelot** [comment]
I imported the theme in the react app
import "primereact/resources/themes/lara-light-cyan/theme.css";


I may load it directly from the global css anotherway ?

I really don't understand , only primereact components are overloading global font ; 
![image](https://github.com/user-attachments/assets/31013184-40d5-4379-ac41-7e7c01aa32f1)

**Comment by sancelot** [comment, marked as answer]
it seems I managed to do it , overloading ;

 .p-component {
  font-family: "Mulish";
}

**Comment by bloodykheeng** [comment]
just add this in globals.css
in my case im using nextjs 15

```
html {
  font-size: 14px;
}
```

it will reduce the fontsize from the default 16px which equate to 1 rem
to the 14px uve set u can even reduce more to ie 10px

**Comment by bloodykheeng** [comment]
u can use  update

 a simple example of how to use the update method with useFieldArray:

```
import React from 'react';
import { useForm, useFieldArray } from 'react-hook-form';

interface Item {
  name: string;
  quantity: number;
}

export default function FieldArrayUpdateExample() {
  const { control, handleSubmit } = useForm<{ items: Item[] }>({
    defaultValues: {
      items: [
        { name: 'Apple', quantity: 5 },
        { name: 'Banana', quantity: 3 }
      ]
    }
  });

  const { fields, update, append } = useFieldArray({
    control,
    name: 'items'
  });

  const onSubmit = (data: { items: Item[] }) => {
    console.log(data);
  };

  const handleUpdateItem = () => {
    // Update the first item (index 0) with new values
    update(0, { 
      name: 'Updated Apple', 
      quantity: 10 
    });
  };

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      {fields.map((field, index) => (
        <div key={field.id}>
          <input 
            defaultValue={field.name} 
            readOnly 
          />
          <input 
            defaultValue={field.quantity} 
            readOnly 
          />
        </div>
      ))}
      
      <button type="button" onClick={handleUpdateItem}>
        Update First Item
      </button>
      
      <button 
        type="button" 
        onClick={() => append({ name: 'New Item', quantity: 1 })}
      >
        Add Item
      </button>
      
      <button type="submit">Submit</button>
    </form>
  );
}
```



In this example:

update(index, newValue) replaces the entire object at the specified index
It takes two arguments:

The index of the item to update
The new object to replace the existing one


You can use it to completely replace an item in the field array
Works well for complex nested objects or simple replacements

The key methods demonstrated:

update: Replace an entire item at a specific index
append: Add a new item to the end of the array
fields: Provides the current array of items with unique IDs
